src/test/scala/com/dwolla/sqlserver/init/RoundTripIntegrationTest.scala

```diff
@@ -0,0 +1,260 @@
+      password <- Gen.const(BuildInfo.test_databasePassword).map(MasterDatabasePassword(_))
+      secrets <- {
+        implicit val arbA: Arbitrary[A] = Arbitrary(genA(database.some))
+        implicit val arbSecret: Arbitrary[Resource[F, SecretId]] = Arbitrary(genSecret[F, A](secretsAlg))
+
+        genResources[F, SecretId]()
+      }
+    } yield secrets.map(DatabaseMetadata(host, port, database, username, password, _, trustServerCert = true))
+  }
+
+  /**
+   * Secrets Manager secrets cost $0.40 / month, so it could be expensive to run this using the real
+   * Secrets Manager service. That said, if you want to run the test that way, replace the
+   * {{{FakeSecretsManagerAlg}}} with {{{SecretsManagerAlg.resource[IO]}}}.
+   */
+  private val secretsManagerAlg: Fixture[SecretsManagerAlg[IO]] = ResourceSuiteLocalFixture(
+    "SecretsManagerAlg",
+    Ref.in[Resource[IO, *], IO, Map[SecretId, String]](Map.empty).map(new FakeSecretsManagerAlg(_))
+  )
+
+  private implicit val prettyRequestResource: Resource[IO, CloudFormationCustomResourceRequest[DatabaseMetadata]] => Pretty = {
+    case Resource.Pure(req) => Pretty(_ => req.asJson.spaces2)
+    case other => Pretty { _ =>
+      other.use { req =>
+        req
+          .ResourceProperties
+          .secretIds
+          .traverse(secretsManagerAlg().getSecret(_).flatMap(parser.parse(_).liftTo[IO]))
+          .map { secrets =>
+            req.applyLens(requestTypeLens)
+              .set(CloudFormationRequestType.OtherRequestType("template"))
+              .asJson
+              .mapObject(_.add("Materialized Secrets (this key is synthetic and not part of the AWS protocol)", secrets.asJson))
+              .spaces2
+          }
+      }.unsafeRunSync() // this would normally be illegal, but it's probably fine here since this only happens when a test failed anyway
+    }
+  }
+
+  override def munitFixtures = List(secretsManagerAlg)
+
+  private def requestTypeLens[T]: Lens[CloudFormationCustomResourceRequest[T], CloudFormationRequestType] = GenLens[CloudFormationCustomResourceRequest[T]](_.RequestType)
+  private val requestDatabaseName: Lens[DatabaseMetadata, Database] = GenLens[DatabaseMetadata](_.name)
+  private val requestDatabaseMetadata: Lens[CloudFormationCustomResourceRequest[DatabaseMetadata], DatabaseMetadata] = GenLens[CloudFormationCustomResourceRequest[DatabaseMetadata]](_.ResourceProperties)
+  private def viaRequestTemplate[T]: Getter[CloudFormationRequestType => CloudFormationCustomResourceRequest[T], CloudFormationCustomResourceRequest[T]] = Getter(_(CloudFormationRequestType.OtherRequestType("n/a")))
+  private val databaseMetadataFromRequestTemplate = viaRequestTemplate.composeLens(requestDatabaseMetadata)
+  private val databaseNameFromRequestTemplate = databaseMetadataFromRequestTemplate.composeLens(requestDatabaseName)
+
+  private implicit val logHandler: LogHandler = LogHandler { event =>
```

Comment:
Did we want to leave the logger configured this way? I noticed it outputs correctly in gha, as well as locally.

Reply:
Probably ok for the integration test but we don't want it logging in prod because it will output passwords